projects/packages/my-jetpack/_inc/components/product-cards-section/index.tsx

```diff
@@ -90,17 +92,52 @@ const DisplayItems: FC< DisplayItemsProps > = ( { slugs } ) => {
 
 interface ProductCardsSectionProps {
 	noticeMessage: ReactNode;
+	siteIsRegistered?: boolean;
 }
 
-const ProductCardsSection: FC< ProductCardsSectionProps > = ( { noticeMessage } ) => {
-	const { ownedProducts = [], unownedProducts = [] } =
+const ProductCardsSection: FC< ProductCardsSectionProps > = ( {
+	noticeMessage,
+	siteIsRegistered,
+} ) => {
+	const { ownedProducts: inntialOwnedProducts = [], unownedProducts: inntialUnownedProducts = [] } =
 		getMyJetpackWindowInitialState( 'lifecycleStats' );
+	const [ siteOwnedProducts, setSiteOwnedProducts ] = useValueStore(
+		'siteOwnedProducts',
+		inntialOwnedProducts
+	);
+	const [ siteUnownedProducts, setSiteUnownedProducts ] = useValueStore(
+		'siteUnownedProducts',
+		inntialUnownedProducts
+	);
+
+	const {
+		data: productOwnershipData,
+		isLoading,
+		refetch: refetchOwnershipData,
+	} = useProductsByOwnership();
+
+	useEffect( () => {
+		refetchOwnershipData();
```

Comment:
Won't it trigger refetches all the time the dependencies of the hook are changed? can we somehow prevent it?

Reply:
I moved it under `!isLoading` and `siteIsRegistered`. It doesn't make sense to run this before the registration status changes. The site is reloaded when someone disconnects it, so we don't need it the other way.